### ISSUE TITLE
diffs didn't match during apply.

### ISSUE THREAD

**Issue by greenhua** [terraform-providers/terraform-provider-google#962]
### Terraform Version

```
    Terraform Version: 0.10.6, 0.11.2
```

### Terraform Configuration Files

```hcl
provider "google" {
  credentials = "${file("gcp.json")}"
  project     = "k-sandbox-1"
  region      = "us-central1"
}

resource "random_string" "gossip" {
  length = 16
  keepers = {
    keeper = "123"
  }

}



resource "google_storage_bucket_object" "gossip" {
  name   = "var.app-group-name-var.app-name-var.env-name-gossip"
  content = "${random_string.gossip.result}"
  bucket = "greenh-consul-store"
  content_type = "text/plain;"
  depends_on = [ "random_string.gossip"]
}
```

### Debug Output

https://gist.github.com/greenhua/de70e09a50da626398c0968785b599a0

### Crash Output

https://gist.github.com/greenhua/fb2361843e61cef13a28a2c81a15d4cb

### Expected Behavior

We need to put file with random string in the GCP bucket

### Actual Behavior

 Diff error returned.

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

### Additional Context
First run "terraform apply" are returning error, second run - it are executing  properly.
diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.

**Comment by danawillow** [comment]
Hi @greenhua, I'm having trouble reproducing this. Can you confirm for me that you can? Here's what my config looks like:

```
provider "google" {}

resource "random_string" "gossip" {
  length = 16
  keepers = {
    keeper = "123"
  }
}
resource "google_storage_bucket" "bucket" {
  name     = "sadfjkdslfadsadsuhtiaer"
}

resource "google_storage_bucket_object" "gossip" {
  name   = "test-bucket-object"
  content = "${random_string.gossip.result}"
  bucket = "${google_storage_bucket.bucket.name}"
  content_type = "text/plain;"
  depends_on = [ "random_string.gossip"]
}
```

And here's the output:
```
$ terraform apply

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + google_storage_bucket.bucket
      id:             <computed>
      force_destroy:  "false"
      location:       "US"
      name:           "sadfjkdslfadsadsuhtiaer"
      project:        <computed>
      self_link:      <computed>
      storage_class:  "STANDARD"
      url:            <computed>

  + google_storage_bucket_object.gossip
      id:             <computed>
      bucket:         "sadfjkdslfadsadsuhtiaer"
      content:        "${random_string.gossip.result}"
      content_type:   "text/plain;"
      crc32c:         <computed>
      detect_md5hash: "different hash"
      md5hash:        <computed>
      name:           "test-bucket-object"
      storage_class:  <computed>

  + random_string.gossip
      id:             <computed>
      keepers.%:      "1"
      keepers.keeper: "123"
      length:         "16"
      lower:          "true"
      number:         "true"
      result:         <computed>
      special:        "true"
      upper:          "true"


Plan: 3 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

google_storage_bucket.bucket: Creating...
  force_destroy: "" => "false"
  location:      "" => "US"
  name:          "" => "sadfjkdslfadsadsuhtiaer"
  project:       "" => "<computed>"
  self_link:     "" => "<computed>"
  storage_class: "" => "STANDARD"
  url:           "" => "<computed>"
random_string.gossip: Creating...
  keepers.%:      "" => "1"
  keepers.keeper: "" => "123"
  length:         "" => "16"
  lower:          "" => "true"
  number:         "" => "true"
  result:         "" => "<computed>"
  special:        "" => "true"
  upper:          "" => "true"
random_string.gossip: Creation complete after 0s (ID: <DKhJP0C)XxnWf=Z)
google_storage_bucket.bucket: Creation complete after 1s (ID: sadfjkdslfadsadsuhtiaer)
google_storage_bucket_object.gossip: Creating...
  bucket:         "" => "sadfjkdslfadsadsuhtiaer"
  content:        "" => "<DKhJP0C)XxnWf=Z"
  content_type:   "" => "text/plain;"
  crc32c:         "" => "<computed>"
  detect_md5hash: "" => "different hash"
  md5hash:        "" => "<computed>"
  name:           "" => "test-bucket-object"
  storage_class:  "" => "<computed>"
google_storage_bucket_object.gossip: Creation complete after 1s (ID: sadfjkdslfadsadsuhtiaer-test-bucket-object)

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

$ terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

random_string.gossip: Refreshing state... (ID: <DKhJP0C)XxnWf=Z)
google_storage_bucket.bucket: Refreshing state... (ID: sadfjkdslfadsadsuhtiaer)
google_storage_bucket_object.gossip: Refreshing state... (ID: sadfjkdslfadsadsuhtiaer-test-bucket-object)

------------------------------------------------------------------------

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.
```

**Comment by greenhua** [comment]
Hm... it is very  interesting. Your code is working without problem but with one difference: your terraform is building new bucket. My should use existing. If you remove bucket creation you can get error.

**Comment by danawillow** [comment]
@greenhua it doesn't make a difference for me whether I create the bucket in Terraform or use an existing one.

**Comment by zbikmarc** [comment]
Seems to work on google provider version `1.5`

**Comment by danawillow** [comment]
Hey @greenhua, can you confirm whether you're still having the issue? Thanks!

**Comment by rosbo** [comment]
Closing since it seems that the issue has been resolved with version 1.5. Please reopen if you are still experiencing this issue.

Thank you
